go.mod

```diff
@@ -3,12 +3,13 @@ module go.einride.tech/protoc-gen-typescript-http
 go 1.17
 
 require (
-	google.golang.org/genproto v0.0.0-20201214200347-8c77b98c765d
-	google.golang.org/protobuf v1.28.1
-	gotest.tools/v3 v3.4.0
+	go.einride.tech/aip v0.67.1
```

Comment:
Can we avoid this dependency? Maybe the parts of the fieldbehaviour package you are using here can be lifted here to avoid depending on an entire 3rd-party library for potentially very little code?

Reply:
On second thought, I am not a 100% well versed in the library but it seems to me that so far the library has only taken into account Protobuf semantics and you are introducing AIP semantics into the code generation.

That'd be mixing 2 different semantics into a library which is potentially only meant to handle Protobuf generation (again not a 100% certain without digging into the details). Is this something we want to do? Nowhere else have I seen this be the case.

Reply:
So I'm fine with avoiding the AIP package because the dependency indeed makes it a little bit convoluted.
However, fieldbehaviours themselves are part of the standard Proto packages (See 
https://pkg.go.dev/google.golang.org/genproto/googleapis/api/annotations#FieldBehavior)

So at that point the within AIP is simply doing the over the annotations field, through the following code.
```
// Get returns the field behavior of the provided field descriptor.
func Get(field protoreflect.FieldDescriptor) []annotations.FieldBehavior {
	if behaviors, ok := proto.GetExtension(
		field.Options(), annotations.E_FieldBehavior,
	).([]annotations.FieldBehavior); ok {
		return behaviors
	}
	return nil
}

// Has returns true if the provided field descriptor has the wanted field behavior.
func Has(field protoreflect.FieldDescriptor, want annotations.FieldBehavior) bool {
	for _, got := range Get(field) {
		if got == want {
			return true
		}
	}
	return false
}
```

In my opinion, this doesn't have anything to do with AIP but is rather a consequence of usage of utility functions that we use in order to build the AIP functionality. So the options at that point are:
1. Copy the AIP Utility functions into this repo to separate the dependencies.
2. Link towards AIP

The semantics are still consistently non-aip, but I think 1) clarifies that.

Let me know what you both thnk

Reply:
As I understand the fieldbehaviour is something Google themselves define and use under their APIs (its under the googleapis package) and it is not defined in the protobuf spec directly. But regardless, the protobuf optional and the annotation OPTIONAL are 2 different things semantically. This PR would essentially treat them the same for the purposes of code generation.

I am not against this change, just mostly arguing for consistency on how I have seen this handled elsewhere